visionmetrics/grounding/recall.py

```diff
@@ -39,6 +41,16 @@ def update(self, predictions, targets):
                      [([target_phrase1, target_phrase2, ...], [[target_bbox, target_bbox, ...], [target_bbox, target_bbox, ...], ...]), ...], type: list of tuple.
                      "target_bboxes" is a list of list of bbox (top, left, bottom, right) in absolute scale.
         """
+        if (len(predictions) != len(targets)):
+            raise ValueError(f"Number of predictions and targets must be equal, got {len(predictions)} and {len(targets)}.")
+        for pred, target in zip(predictions, targets):
```

Comment:
If this grounding specific, shouldn't it be enough for the ground truth wrt text being the text span instead of a real string?

Reply:
Ideally yes, the string itself does not matter if the text span matches.